### PR TITLE
Transform store and authSlice from JS to TS

### PR DESCRIPTION
**Pull request by wariofdz**
### Changed:

 - Transformed `store` from JS to TS.
 - Transformed `authSlice` and his test from JS to TS.
 - Transformed `createAuthByCredentials` and his test from JS to TS.
 - Transformed `createAuthByToken` and his test from JS to TS.
 - Updated `buildSerializableResponse` add a new type `SerializableResponseUnion`.
 - Added new file `hooks.ts` to implement `useSelector` and `useDispatch` as it recommends Redux website.
 - Added new types `AuthResponse`, `AuthSerializedResponse`, `SerializableResponseUnion`, `SerializedResponse`. 
 - Added new enum `AuthStateStatus`.
 - Added new interface `BaseAuthState` and new types `NonAuthenticatedAuthState`, `PendingAuthState`, `AuthenticatedAuthState` on AuthState file.
 - Changed on `createAuthByCredentials`, add new interface `CreateAuthByCredentialsParams`.
 - Changed on `createInitialState` function and his test.
 - Changed on `authSlice` and his test.
 - Deleted `authSliceStatus` file.

### CODE REVIEW

File: packages/frontend/web-ui/src/app/store/features/authSlice.spec.js [hunk 1]
```diff
@@ -129,7 +129,7 @@ describe('authSlice', () => {
     beforeAll(() => {
```
Comment (notaphplover) [inline packages/frontend/web-ui/src/app/store/features/authSlice.spec.js:129]: This file should be migrated as well

**Review by notaphplover — CHANGES_REQUESTED** [review]
Nicely done! Let's migrate that test and merge it

File: packages/frontend/web-ui/src/app/store/features/authSlice.spec.js (hunk 1 above)
Comment (wariofdz) [reply packages/frontend/web-ui/src/app/store/features/authSlice.spec.js]: Done! But... I'm still working with the `createAuthByToken` and `createAuthByCredentials` tests to convert them into TS.

Comment (wariofdz) [reply packages/frontend/web-ui/src/app/store/features/authSlice.spec.js]: Done in both tests!

**Review by notaphplover — APPROVED** [review]

**Pull request merged**